tasks/opensource/install-redhat.yml

```diff
@@ -1,26 +1,82 @@
 ---
 - name: (AlmaLinux/Amazon Linux/CentOS/Oracle Linux/RHEL/Rocky Linux) {{ (nginx_setup == 'uninstall') | ternary('Remove', 'Configure') }} NGINX repository
+  vars:
+    distribution: "{{ ansible_facts['distribution'] }} {{ ansible_facts['system'] }} {{ ansible_facts['distribution_version'] }}"
   ansible.builtin.yum_repository:
     name: nginx
-    baseurl: "{{ nginx_repository | default(lookup('vars', 'nginx_default_repository_' + ((ansible_facts['distribution'] == 'Amazon') | ternary('amazon', 'redhat')))) }}"
+    baseurl: "{{ nginx_repository | default(lookup('vars', 'nginx_default_repository_' + ((ansible_facts['distribution'] == 'Amazon') | ternary('amazon', 'redhat'))))\
+      \ }}"
     description: NGINX Repository
     enabled: true
     gpgcheck: true
     mode: "0644"
     module_hotfixes: true
     state: "{{ (nginx_state == 'uninstall') | ternary('absent', 'present') }}"
-  when: nginx_manage_repo | bool
+  when:
+    - nginx_manage_repo | bool
+    - distribution != 'Amazon Linux 2023'
+
+- name: (Amazon Linux 2023) {{ (nginx_setup == 'uninstall') | ternary('Remove', 'Configure') }} NGINX stable repository
```

Comment:
Check what's being done behind the scenes in the vars file. I'd like to keep everything contained within the task that already exists.